pandas/tests/frame/methods/test_sort_values.py

```diff
@@ -19,7 +19,7 @@ def test_sort_values_sparse_no_warning(self):
         # GH#45618
         # TODO(2.0): test will be unnecessary
         ser = pd.Series(Categorical(["a", "b", "a"], categories=["a", "b", "c"]))
-        df = pd.get_dummies(ser, sparse=True)
+        df = pd.get_dummies(ser, dtype=np.uint8, sparse=True)
```

Comment:
🤔 I wouldn't have thought this needed to change, will take a closer look

Reply:
so, the warning is raised in

https://github.com/pandas-dev/pandas/blob/576302378541230f63a0ad782ed8df399f488a00/pandas/core/arrays/sparse/array.py#L1328-L1338

because with `dtype=bool`, in 

https://github.com/pandas-dev/pandas/blob/576302378541230f63a0ad782ed8df399f488a00/pandas/core/indexes/base.py#L5213-L5215

then with `np.uint8` we have:
```
(Pdb) p self
UInt64Index([0, 1], dtype='uint64')
```

whereas with `bool` we have:
```
(Pdb) p self
Index([False, True], dtype='Sparse[bool, False]')
```

this'll need closer examination

Reply:
What is your suggestion? 🤔

Reply:
don't know yet, will have a look

Reply:
seems related to https://github.com/pandas-dev/pandas/issues/45618

I'll see if I can come up with a separate reproducer, might just be the case of changing

https://github.com/pandas-dev/pandas/blob/6b93a0c6687ad1362ec58715b1751912783be411/pandas/core/sorting.py#L347

to 

```
warnings.filterwarnings("ignore", ".*(SparseArray|SparseDtype).*", category=FutureWarning)
```

(that'd be for a separate PR though - I'll try to do that first)

Reply:
ok done - @kianelbo could you fetch and merge upstream/main, and discard any changes you made to `pandas/tests/frame/methods/test_sort_values.py`?

I think something like
```
git fetch upstream
git merge upstream/main
git checkout upstream/main -- pandas/tests/frame/methods/test_sort_values.py
git add -u
git commit
```
would work